### PR TITLE
New sign in modal

### PR DESCRIPTION
**Pull request by rogerhutchings**
Adds in the new auth modal designs for login, uses Formik for form handling.

Due to the impending deadline, I'm going to add any remaining tests in a future PR.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

File: packages/app-project/src/components/AuthModal/AuthModal.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    </Modal>
+  )
+}
+
+AuthModal.propTypes = {
+  className: string,
+  closeModal: func.isRequired,
+}
+
+export default withOnlyRenderOnBrowser(withThemeContext(AuthModal, authModalTheme))
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/AuthModal.js]: We could wrap the `Modal` component directly with `withOnlyRenderOnBrowser` in the lib-react-components (and remove the check I had added that doesn't actually work). That's what I had done in the close PR where I was attempting updates on these... We'll probably always want this check until the underlying Layer becomes SSR friendly.

File: packages/app-project/src/components/AuthModal/AuthModal.spec.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { mount, shallow} from 'enzyme'
+import React from 'react'
+
+import AuthModal from './AuthModal'
+
+const CLOSE_MODAL = Function.prototype
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/AuthModal.spec.js]: Missing test for this.

File: packages/app-project/src/components/AuthModal/AuthModal.spec.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    })
+  })
+
+  it('should render without crashing', function () {
+    expect(wrapper).to.be.ok()
+  })
+
+  it('should pass through the `activeIndex` and `onActive` props to Tabs', function () {
+    // `Tabs` should be the first component
+    const tabsWrapper = wrapper.children().first()
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/AuthModal.spec.js]: This variable isn't being used.

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import React from 'react'
+
+import en from './locales/en'
+
+counterpart.registerTranslations('en', en)
+
+const userNameFieldId = 'LoginForm_login'
+const passwordFieldId =  'LoginForm_password'
+
+const textColors = { dark: 'white', light: 'black' }
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js]: I don't think this is necessary. Looking at the design, it looks like the text color is the standard we've already set in the theme for light, #5C5C5C. @beckyrother could you confirm or make https://projects.invisionapp.com/share/UAN6AP0JTBC#/screens/365296253_00-31_Sign_In inspectable? It's only the share link.

File: packages/app-project/src/components/AuthModal/AuthModal.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    >
+      <StyledTabs activeIndex={activeIndex} onActive={onActive}>
+        <Tab title='Sign In'>
+          <Box width='medium'>
+            <LoginForm closeModal={closeModal} />
+          </Box>
+        </Tab>
+        <Tab title='Register'>
+          <Box width='large'>
+            bar
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/AuthModal.js]: This should probably just be hidden if it's not ready yet.

File: packages/app-project/src/components/AuthModal/AuthModal.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { Modal, Tabs, Tab, withThemeContext } from '@zooniverse/react-components'
+import { Box } from 'grommet'
+import { func, string } from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+
+import authModalTheme from './theme'
+import LoginForm from './components/LoginForm'
+import withOnlyRenderOnBrowser from './shared/components/withOnlyRenderOnBrowser'
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/AuthModal.js]: This import hasn't been switch to the one moved into lib-react-components, so we have duplicate code now.

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginFormContainer.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+@inject('store')
+@observer
+class LoginFormContainer extends Component {
+  constructor () {
+    super()
+    this.onSubmit = this.onSubmit.bind(this)
+  }
+
+  onSubmit (values, { setFieldError, setSubmitting }) {
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/components/LoginForm/LoginFormContainer.js]: We could add a test for the functionality of this handler this using nock.

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        }) => (
+          <Box as='form' onSubmit={handleSubmit} margin={{ top: 'small' }}>
+
+            <FormField
+              error={errors.login && touched.login && errors.login}
+              htmlFor={userNameFieldId}
+              label={<Label>Username or email address</Label>}
+            >
+              <TextInput
+                disabled={isSubmitting}
```
Comment (srallen) [inline packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js]: We have a regression here on the focus functionality... Just tried `autoFocus` again here, and this time it worked, so I'm not 100% why it was breaking in my branch and PR.

File: packages/app-project/src/components/AuthModal/AuthModal.spec.js (hunk 2 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/AuthModal.spec.js]: Added via #888

File: packages/app-project/src/components/AuthModal/AuthModal.spec.js (hunk 3 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/AuthModal.spec.js]: Fixed via #888

File: packages/app-project/src/components/AuthModal/AuthModal.js (hunk 5 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/AuthModal.js]: Fixed via #888

File: packages/app-project/src/components/AuthModal/AuthModal.js (hunk 6 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/AuthModal.js]: Fixed via #888

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js (hunk 8 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js]: Fixed via #888

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js (hunk 4 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js]: I've set this to `light-1` / `dark-5` - without it, all text renders as black in both light and dark modes.

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginFormContainer.js (hunk 7 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/components/LoginForm/LoginFormContainer.js]: I've written tests for this using sinon - we're not testing the auth client, just that it's called correctly

File: packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js (hunk 4 above)
Comment (rogerhutchings) [reply packages/app-project/src/components/AuthModal/components/LoginForm/LoginForm.js]: also, cc'ing #890

**Comment by srallen** [comment]
Merging so we have a semi-working registration button before TESS launch.

**Comment by beckyrother** [comment]
Sorry just saw the invision request. Were you looking for design signoff before this was merged?

**Comment by srallen** [comment]
Let's do design review for #889. Unfortunately this one had to get merged ASAP.

**Pull request merged**